.github/workflows/func.yaml

```diff
@@ -171,11 +174,21 @@ jobs:
            else
              echo "Skip because missing openstack-auth-env secrets"
            fi
+      - name: pytest env
+        if: ${{ inputs.pytest-env }}
+        run: |
+           echo ${{ inputs.pytest-env }} | base64 -d | grep 'PYTEST_' >> "$GITHUB_ENV"
+      - name: zaza yaml
+        if: ${{ inputs.zaza-yaml }}
+        run: |
+           echo ${{ inputs.zaza-yaml }} | base64 -d >> $HOME/.zaza.yaml
       - name: external controller required package
         if: ${{ inputs.external-controller }}
         run: |
+           # unzip is needed because some of the legacy charm require unzip in Makefile
+           sudo apt install -y unzip
            sudo snap install charmcraft --channel latest/stable --classic
-           sudo snap install juju --channel ${{ inputs.juju-channel }}
+           sudo snap install juju --channel ${{ inputs.juju-channel }} --classic
```

Comment:
Only `2.9/stable` requires the `--classic` flag, not `3.1/stable`. Not sure how it would impact the installation and usage there.

Reply:
Even if we pass the `--classic` flag in juju 3, it will install as usual. So I think keeping this flag is okay to make it compatible across both 2.9 and 3.x.